src/tensor_ops/sum_to/cuda_kernel.rs

```diff
@@ -1,27 +1,89 @@
 use crate::{
-    shapes::{Axes, Dtype, ReduceShapeTo, Shape},
-    tensor::Cuda,
+    shapes::{Axes, BroadcastStridesTo, ReduceShapeTo, Shape},
+    tensor::cuda::{Cuda, CudaArray},
 };
 
-impl<E: Dtype> super::SumKernel<E> for Cuda {
+use cudarc::device::{CudaSlice, LaunchAsync, LaunchConfig};
+
+use std::sync::Arc;
+
+const MODULE_NAME: &str = "sum_to";
+const FWD_FN_NAME: &str = "sum_to_forward";
+const BWD_FN_NAME: &str = "sum_to_backward";
+const ALL_FN_NAMES: [&str; 2] = [FWD_FN_NAME, BWD_FN_NAME];
+const PTX_SRC: &str = include_str!(concat!(env!("OUT_DIR"), "/sum_to.ptx"));
+
+impl super::SumKernel<f32> for Cuda {
     fn forward<Src: Shape, Dst: Shape, Ax: Axes>(
         &self,
         dst: Dst,
-        inp: &Self::Storage<Src, E>,
-    ) -> Result<Self::Storage<Dst, E>, Self::Err>
+        inp: &Self::Storage<Src, f32>,
+    ) -> Result<Self::Storage<Dst, f32>, Self::Err>
     where
         Src: ReduceShapeTo<Dst, Ax>,
     {
-        todo!()
+        if !self.dev.has_func(MODULE_NAME, FWD_FN_NAME) {
+            self.dev
+                .load_ptx(PTX_SRC.into(), MODULE_NAME, &ALL_FN_NAMES)?;
+        }
+
+        let fwd_fn = self.dev.get_func(MODULE_NAME, FWD_FN_NAME).unwrap();
+
+        let dims: CudaSlice<usize> = self.dev.take_async(inp.shape.concrete().into())?;
+        let inp_strides: CudaSlice<usize> = self.dev.take_async(inp.strides.into())?;
+        let out_strides: Src::Concrete =
+            BroadcastStridesTo::<Src, Ax>::broadcast_strides(&dst, dst.strides());
```

Comment:
the only difference between normal impl of forward and this is this call where the strides are broadcast. Maybe there is some way aroudn that?